PublicFolders/ValidateEXOPFDumpster.ps1

```diff
@@ -0,0 +1,423 @@
+        #Validate till Non_Ipm_Subtree
+        if ($PublicFolderInfo.Publicfolder.ParentFolder -ne  $PublicFolderInfo.NON_IPM_SUBTREE.EntryId) {
+            $ParentPublicFolderInfo=GetPublicFolderInfo($PublicFolderInfo.Publicfolder.ParentFolder)
+            ValidateContentMBXUniqueness($ParentPublicFolderInfo)
+            ValidateEntryIDMapping($ParentPublicFolderInfo)
+            ValidateContentMBXQuota($ParentPublicFolderInfo)
+            ValidateParentPublicFolder($ParentPublicFolderInfo)
+        }
+    }
+    #Validate on IPM_Subtree
+    else {
+        if ($PublicFolderInfo.Publicfolder.ParentFolder -ne  $PublicFolderInfo.IPM_SUBTREE.EntryId) {
+            $ParentPublicFolderInfo=GetPublicFolderInfo($PublicFolderInfo.Publicfolder.ParentFolder)
+            ValidateContentMBXUniqueness($ParentPublicFolderInfo)
+            ValidateEntryIDMapping($ParentPublicFolderInfo)
+            ValidateContentMBXQuota($ParentPublicFolderInfo)
+            ValidateParentPublicFolder($ParentPublicFolderInfo)
+        }
+    }
+}
+Function ValidateDumpsterFlag {
+    param([PSCustomObject]$PublicFolderInfo)
+    if ($PublicFolderInfo.Publicfolder.AdminFolderFlags -eq "DumpsterFolder") {
+        $Issue="Public folder $($PublicFolderInfo.Publicfolder.Identity) is a dumpster folder, content folder and its dumpster are linked to each other, that link cannot be undone, in other words admins cannot delete dumpster only"
+        $Fix="FIX --> Please move $($PublicFolderInfo.Publicfolder.Identity) public folder to be under NON_IPM_SUBTREE using Set-PublicFolder command, for more information please check https://aka.ms/setpf"
+        WritetoScreenANDlog -Issue $Issue -Fix $Fix
+    }
+}
+Function GetUserPermissions {
+    Param([PSCustomObject]$Perms)
+    $workingpermissions=@("editor", "owner", "publishingeditor", "deleteallitems")
+    if ($null -ne $Perms) {
+        foreach ($perm in $Perms.AccessRights) {
+            if ($workingpermissions.Contains($($perm.ToLower()))) {
+                return "user has permission"
+            }
+        }
+        return "user has no permission"
+    }
+
+    else {
+        return "user has no permission"
+    }
+}
+Function ValidateUserPermissions {
+    Param([PSCustomObject]$Perms, [PSCustomObject]$Permsuserpfmbx)
+    if ($Perms.AccessRights.count -eq $Permsuserpfmbx.AccessRights.count) {
+        foreach ($Perm in $perms.AccessRights) {
+            if (!$Permsuserpfmbx.AccessRights.contains($Perm)) {
+                return "sync issue exist"
+            }
+        }
+        return "no sync issue exist"
+    }
+    return "sync issue exist"
+}
+Function ValidatePublicFolderIssue {
+    param([PSCustomObject]$PublicFolderInfo)
+    $ItemORFolder=Read-Host "Please specify if the issue is related to a user who is not able to delete an item inside a public folder or neither a user with owner permissions nor the admin are not able to delete the Public folder as awhole, Type (I) for Item or (F) for Folder"
```

Comment:
typo - awhole

Reply:
Fixed and installed the recommended plugin Thanks!